source/lac/sparsity_tools.cc

```diff
@@ -619,22 +616,21 @@ namespace SparsityTools
           MPI_Probe(MPI_ANY_SOURCE, MPI_ANY_TAG, mpi_comm, &status);
           Assert (status.MPI_TAG==124, ExcInternalError());
 
-          MPI_Get_count(&status, MPI_BYTE, &len);
-          Assert( len%sizeof(typename DSP_t::size_type)==0, ExcInternalError());
-
-          recv_buf.resize(len/sizeof(typename DSP_t::size_type));
-
-          MPI_Recv(&recv_buf[0], len, MPI_BYTE, status.MPI_SOURCE,
+          MPI_Get_count(&status, DEAL_II_DOF_INDEX_MPI_TYPE, &len);
+          recv_buf.resize(len);
+          MPI_Recv(&recv_buf[0], len, DEAL_II_DOF_INDEX_MPI_TYPE, status.MPI_SOURCE,
                    status.MPI_TAG, mpi_comm, &status);
 
           typename std::vector<typename DSP_t::size_type>::const_iterator ptr = recv_buf.begin();
           typename std::vector<typename DSP_t::size_type>::const_iterator end = recv_buf.end();
-          while (ptr+1<end)
+          while (ptr!=end)
```

Comment:
I see. I didn't read the code carefully enough to realize that you're skipping over more than just one entry per go-around of the while loop.